Supplier model

igorsergiichuk opened the pull request:
## Overview

| Q                 | A
| ----------------- | ---
| Bugfix?           | no
| New feature?      | yes
| Breaking Changes? | no
| Fixed issues      | #

## The problem

Supplier model needed to be added to the project

## How I resolved it

Created new Supplier Model, only uses basic CRUD so no special methods. Test and mocks are in progress.

## How to verify it

Run the tests

Notify the following people: @darrynten

igorsergiichuk committed:
Supplier Model

Build by I.Serhiichuk on 06.12.2077

igorsergiichuk committed:
Update Supplier.php

All changes due to notes:
src/Models/Supplier.php:4 - one too many newlines
src/Models/Supplier.php:23 - No newline after class opening
src/Models/Supplier.php:45 - This docblock contains training information which must be replaced with actual information
src/Models/Supplier.php:320 - 1 too many newlines

darrynten commented on src/Models/Supplier.php:
```diff
@@ -0,0 +1,306 @@
+        'save' => true,
+        'delete' => true,
+    ];
+
```
This is where StyleCI is failing :)

igorsergiichuk replied:
I've already fixed that.) Thanks. The only thing I need to pull is review.

igorsergiichuk committed:
Update Supplier.php

darrynten commented on src/Models/Supplier.php:
```diff
@@ -0,0 +1,305 @@
+
+/**
+ * Supplier Model
+ *
```
You need to add some information here, and also leave a link to the documentation for this model

igorsergiichuk committed:
Update Supplier.php

Due to the notes by Darryn

igorsergiichuk reviewed (commented):
Added a link to Supplier parameters:
https://accounting.sageone.co.za/api/1.1.2/Help/Api/POST-Supplier-Save

darrynten commented on src/Models/Supplier.php:
```diff
@@ -0,0 +1,307 @@
+ * Supplier Model
+ *
+ * Details on parameters for Supplier:
+ * https://accounting.sageone.co.za/api/1.1.2/Help/Api/POST-Supplier-Save
```
This link should be https://accounting.sageone.co.za/api/1.1.2/#bookmark_Supplier

The link you attached for is specific to the save method

igorsergiichuk committed:
Update Supplier.php

mxnr reviewed (dismissed):
Looks good for me👍

darrynten commented:
@mxnr it's looking good, we're just waiting on the tests and mocks now :) But I agree, @igorsergiichuk is doing a great job here

igorsergiichuk committed:
Changed TestGetAll and TestGetId

Updated in TestGetAll to $this->assertCount(2, $results); from $this->assertEquals(2, count($results));
and
In TestGetId set ID 11 (verifyGetId(Account::class, 11, function ($model)) same as in mock

darrynten commented on tests/SageOne/Models/AccountModelTest.php:
```diff
@@ -176,7 +176,7 @@ public function testFeatures()
     {
         $this->verifyGetAll(Account::class, function ($results) {
-            $this->assertEquals(2, count($results));
+            $this->assertCount(2, $results);
```
Thanks for changing this

darrynten commented on src/Models/Supplier.php:
```diff
@@ -0,0 +1,307 @@
+            'readonly' => false,
+        ],
+        'category' => [
+            'type' => 'SupplierCategory',
```
Does this model exist yet? If it doesn't then we will need to add it as part of this PR

darrynten commented:
Superseded by #30
